src/components/label.js

```diff
@@ -12,49 +13,86 @@ const _labelSize = (size) => {
+const _colorType = (type, color) => {
+  let colorType = ''
+
+  switch(type) {
+  case 'light':
+    const borderElementsColor = concatStateColor(color, 'shadow', ['focus', 'hover'], [500])
+    const buttonColor = concatTargetColor(color, ['bg', 'shadow'], [100, 100, 500])
+    const textColor = concatTargetColor('secondary', ['text'], [900])
+
+    colorType = `${textColor} ${buttonColor} ${borderElementsColor}`
+    break
+  case 'dark':
+    if (color === 'warning') {
+      const borderElementsColor = concatStateColor(color, 'shadow', ['focus', 'hover'], [500])
+      const buttonColor = concatTargetColor(color, ['bg', 'shadow'], [100, 100, 500])
+      const textColor = concatTargetColor('secondary', ['text'], [900])
+
+      colorType = `${textColor} ${buttonColor} ${borderElementsColor}`
+    } else {
+      const borderElementsColor = concatStateColor(color, 'shadow', ['focus', 'hover'], [500])
+      const buttonColor = concatTargetColor(color, ['bg', 'shadow'], [100, 100, 500])
+      const textColor = concatTargetColor('secondary', ['text'], [50])
+
+      colorType = `${textColor} ${buttonColor} ${borderElementsColor}`
+    }
+    break
+  default:
+    break
+  }
+  
+  return colorType
+}
+
+const customClasses = makeStyles({
+  buttonContainer: {
+    borderWidth: 0
+  }
+})
+
+const Label = forwardRef(({ classes, children, startIcon, endIcon, title, size, shade, color, selectable, ...rest }, ref) => {
   const labelSize = _labelSize(size)
-  const borderElementsColor = concatStateColor(color, 'border', ['focus', 'hover'], [500])
-  const buttonColor = concatTargetColor(color, ['bg', 'border', 'text'], [100, 100, 500])
-  console.log('label')
+  const colorType = _colorType(shade, color)
+
   const labelClasses = Object.freeze({
-    button: `px-5px rounded-md  ${classes.root ? classes.root : ''}
-      border fill-current ${buttonColor} focus:outline-none ${borderElementsColor}
+    button: `rounded-3px  ${classes.root ? classes.root : ''} ${labelSize.box} ${customClasses.buttonContainer}
+      border fill-current focus:outline-none ${colorType}
       ${selectable ? 'cursor-pointer' : 'pointer-events-none'}
     `, 
-    content: `font-bold ${labelSize.font} ${classes.content ? classes.content : ''}`, 
+    content: `font-bold tracking-normal leading-1 ${labelSize.font} ${classes.content ? classes.content : ''}`, 
     title: `font-normal`,
-    startIcon: `mr-5px ${classes.startIcon ? classes.startIcon : ''}`, 
-    endIcon: `ml-5px ${classes.endIcon ? classes.endIcon : ''}`,
+    startIcon: `mr-1 ${classes.startIcon ? classes.startIcon : ''}`, 
+    endIcon: `ml-1 ${classes.endIcon ? classes.endIcon : ''}`,
   })
 
   return (
     <ButtonBase ref={ref} classes={labelClasses} startIcon={startIcon} endIcon={endIcon} disabled={!selectable} {...rest}>
-      {children} <span>{title}</span>
+      {children}{title && <>: <span className={`${labelClasses.title}`}>{title}</span></>}
```

Comment:
how do we access label class from the outside?

Reply:
you can access each of the main container label classes passing down a prop in this way:
```
classes: { 
    root: 'custom-root', 
    content: 'custom-container', 
    title: 'custom-title', 
    startIcon: 'custom-start-icon', 
    endIcon: 'custom-end-icon' 
  }
```